keel/quotations/admin.py

```diff
@@ -16,7 +28,38 @@ class QuotationAdmin(admin.ModelAdmin):
 
 
 class QuotationMilestioneAdmin(CustomBaseModelAdmin):
-    pass
+    
+    def save_model(self, request, obj, form, change):
+        id = str(obj.pk)
+        customer = obj.quotation.user
+        
+        try:
+            case = Case.objects.get(user=customer)
+        except Exception as e:
+            logger.error('ERROR: QUOTATIONS:QuotationMilestioneAdmin ' + str(e)) 
+            return super().save_model(request, obj, form, change)
+        
+        payment_manager = PaymentManager()
+        data = {
+                "order_items": {
+                    "quotation_milestone":[id],
+                    "plan": [],
+                    "service": []
+                }
+            }
+        try:
+            payment_details = payment_manager.generate_payment_details(
+                StructNewPaymentDetailArgs(customer_id=customer.pk, customer_currency="usd",
+                                           initiator_id=customer.pk, payment_client_type=PAYMENT_CLIENT_TYPE,
+                                           case_id=case),
+                data["order_items"]
+            )
+            print(payment_details)
+        except ValueError as err:
+            logger.error('ERROR: QUOTATIONS:QuotationMilestioneAdmin ' + str(e))
```

Comment:
Use logging_format function so that logging message is uniform throughout our app in keel.Core.err_log here and for severity use LOGGER_CRITICAL_SEVERITY from keel.Core.constants.